tests/suites/test_suite_psa_crypto.function

```diff
@@ -818,7 +818,7 @@ void mac_key_policy( int policy_usage_arg,
                 mbedtls_test_update_key_usage_flags( policy_usage ) );
 
     status = psa_mac_sign_setup( &operation, key, exercise_alg );
-    if( ( policy_usage & PSA_KEY_USAGE_SIGN_HASH ) == 0 )
+    if( ( mbedtls_test_update_key_usage_flags(policy_usage) & PSA_KEY_USAGE_SIGN_MESSAGE ) == 0 )
```

Comment:
I think this change is correct considering the starting point, but I'm afraid I'm unhappy enough with the starting point that I'm going to ask you to change the logic here: we really should have `expected_status_sign_arg` and `expected_status_verify_arg` as arguments. Then we'd have no `if` statement here and a simple `TEST_EQUAL( status, expected_status_sign );`.

The way it's currently done with the if statement is very misleading. When I look at the data file and see a test case like
```
PSA key policy: MAC, SIGN_MESSAGE
depends_on:PSA_WANT_ALG_HMAC:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_HMAC
mac_key_policy:PSA_KEY_USAGE_SIGN_MESSAGE:PSA_ALG_HMAC(PSA_ALG_SHA_256):PSA_KEY_TYPE_HMAC:"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa":PSA_ALG_HMAC(PSA_ALG_SHA_256):PSA_SUCCESS
```
it is extremely surprising to find out that, before your PR, the test actually asserted that the sign and verify operations would both fail with `PSA_ERROR_NOT_PERMITTED` - the test data clear says `PSA_SUCCESS` and this is absolutely ignored by the test function. After your patch, the test data is exactly the same but now the test asserts that the sign operation returns `PSA_SUCCESS` (as is visible from the data file) and that the verify operation returns `PSA_ERROR_NOT_PERMITTED` (which is completely surprising when looking at the data file which only says `PSA_SUCCESS`).

The test data should end with `PSA_SUCCESS:PSA_ERROR_NOT_PERMITTED` - then it's clear that we expect one operation to succeed and the other to fail.

Sorry, I know it's pre-existing, but I'm afraid I can't approve the PR without fixing this, as the test data is still very misleading.

Reply:
Changed mac_key_policy function to accept expected status for both sign and verify operations